app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java

```diff
@@ -67,6 +78,33 @@ public class EventsFragment extends Fragment {
         @Override
         public void onMapReady(GoogleMap googleMap) {
             mMap = googleMap;
+            mMap.setOnMarkerClickListener(new GoogleMap.OnMarkerClickListener() {
+                @Override
+                public boolean onMarkerClick(@NonNull Marker marker) {
+
+                    if (doubleClickPress) {
+                        StoreFront store = mMarkertoStore.get(marker);
+                        Intent i = new Intent(getContext(), DetailActivity.class);
+                        i.putExtra(StoreFront.class.getSimpleName(), Parcels.wrap(store));
+                        getContext().startActivity(i);
+                    } else {
+                        doubleClickPress = true;
+
+                        new Handler().postDelayed(new Runnable() {
+                            @Override
+                            public void run() {
+                                doubleClickPress = false;
+                            }
+                        }, DELAY_TIME);
+                    }
+
+//                    StoreFront store = mMarkertoStore.get(marker);
```

Comment:
Commented code

Reply:
Also, I notice that this code looks more or less identical to the code on 86-89. You might want to refactor it into a new method navigateToStore(store)